src/mapgen/mapgen_v7.cpp

```diff
@@ -440,6 +471,32 @@ int MapgenV7::generateTerrain()
 		noise_mountain->perlinMap3D(node_min.X, node_min.Y - 1, node_min.Z);
 	}
 
+	//// Floatlands
+	// Bool for optimisation of condition checks in y-loop
+	bool gen_floatlands = false;
+	float float_offset_cache[csize.Y + 2];
```

Comment:
This should rather be allocated on the heap.

Reply:
Ok, i have no idea how, but will try to find examples elsewhere in MT code to learn from.

Reply:
like usual: `float *float_offset_cache = new float[csize.Y + 2];` here and `delete[] float_offset_cache;` to free the memory

Reply:
Ok thanks, will do.
(I just did a websearch, which suggested using `malloc()`, and that seemed weird.)

Reply:
I found examples elsewhere in code, but those always allocate in the constructor and delete in the destructor. It possibly seems better to do this here too to avoid allocating and freeing for every mapchunk?

Reply:
That would break `num_emerge_threads > 1` in a subtle way since a single `Mapgen` instance is used in multiple threads.
Allocating every time is not a problem either way, e.g. the client also makes many allocations to render a single frame.

Reply:
Ok thanks.
Allocation done.

Reply:
Sorry, just noticed what I said is wrong. Every thread has its own mapgen, so the caching the noise buffer as a class variable is definitely okay.
(One allocation per chunk is not too much though, so it's not <i>required</i> to cache it)

Reply:
Aha ok, just saw this comment, will do.

Reply:
Now a class variable.